Application.hs

```diff
@@ -47,92 +50,160 @@ import Handler.Demo
 -- comments there for more details.
 mkYesodDispatch "App" resourcesApp
 
--- This function allocates resources (such as a database connection pool),
--- performs initialization and creates a WAI application. This is also the
--- place to put your migrate statements to have automatic database
+-- | This function allocates resources (such as a database connection pool),
+-- performs initialization and return a foundation datatype value. This is also
+-- the place to put your migrate statements to have automatic database
 -- migrations handled by Yesod.
-makeApplication :: AppConfig DefaultEnv Extra -> IO Application
-makeApplication conf = do
-    foundation <- makeFoundation conf
+makeFoundation :: AppSettings -> IO App
+makeFoundation appSettings = do
+    loadEnv
+
+    dbconf <- if appDatabaseUrl appSettings
+        then postgresConf 10
```

Comment:
TODO: use the poolsize from appDatabaseConf, then we can set `PGPOOLSIZE=100` on staging/production. We've been using a hardcode `10` currently anyway, but that'll be a nice fix.